checkov/terraform/checks/provider/base_check.py

```diff
@@ -1,30 +1,22 @@
 from abc import abstractmethod
+from typing import List, Dict, Any
 
 from checkov.common.checks.base_check import BaseCheck
-from checkov.common.multi_signature import multi_signature
+from checkov.common.models.enums import CheckCategories, CheckResult
 from checkov.terraform.checks.provider.registry import provider_registry
 
 
 class BaseProviderCheck(BaseCheck):
-    def __init__(self, name, id, categories, supported_provider):
-        super().__init__(name=name, id=id, categories=categories, supported_entities=supported_provider,
-                         block_type="provider")
+    def __init__(self, name: str, id: str, categories: List[CheckCategories], supported_provider: List[str]) -> None:
+        super().__init__(
+            name=name, id=id, categories=categories, supported_entities=supported_provider, block_type="provider"
+        )
         self.supported_provider = supported_provider
         provider_registry.register(self)
 
-    def scan_entity_conf(self, conf, entity_type):
-        return self.scan_provider_conf(conf, entity_type)
+    def scan_entity_conf(self, conf: Dict[str, List[Any]], entity_type: str) -> CheckResult:
+        return self.scan_provider_conf(conf)
 
-    @multi_signature()
     @abstractmethod
-    def scan_provider_conf(self, conf, provider_type):
+    def scan_provider_conf(self, conf: Dict[str, List[Any]]) -> CheckResult:
         raise NotImplementedError()
-
-    @classmethod
-    @scan_provider_conf.add_signature(args=["self", "conf"])
-    def _scan_provider_conf_self_conf(cls, wrapped):
-        def wrapper(self, conf, provider_type=None):
-            # keep default argument for entity_type so old code, that doesn't set it, will work.
-            return wrapped(self, conf)
-
-        return wrapper
```

Comment:
@gruebel Not sure we want to delete this... Do we?

Reply:
I checked current code base and there is no check, which utilises the old way of handling it with `provider_type`. if someone wants to also do something with the `provider_type`, then he/she can use the method `scan_entity_conf()` instead. Or is there any reason, why you want to keep it inside the code base?

Reply:
There is no check and I find it hard to think of a valid usecase. It seems more relevant for resource checks than provider checks... Let's remove it for now. Thanks!